pymeasure/instruments/minicircuits/usb_2sp2t_dch.py

```diff
@@ -0,0 +1,101 @@
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+from pymeasure.instruments import Instrument, RangeException
+from pymeasure.instruments.validators import truncated_range, strict_discrete_set
+import clr
+clr.AddReference('mcl_SolidStateSwitch_NET45')
+from mcl_SolidStateSwitch_NET45 import USB_Digital_Switch
+
+import numpy as np
+import time
+
+
+class USB_2SP2T_DCH():
+    """ Represents the USB-2SP2T-DCH solid state switch and provides a high level
+    interface to the SCIPI commands. Requires that the mcl_SolidStateSwitch_NET45 be in the path
+    or same directory as the running code and pythonnet is installed to import clr. DLL may be obtained from:
+    https://www.minicircuits.com/softwaredownload/solidstate.html
+
+    """
+
+    def __init__(self, SN = None):
+        self.connection = USB_Digital_Switch()
+        if SN is not None:
+            if isinstance(SN,str):
+                self.connection.Connect(SN)
+            elif isinstance(SN,int):
+                self.connection.ConnectByAddress(SN)
+            else:
+                raise TypeError(f'got type {type(SN)} which is not allowed')
+        else:
+            self.connection.Connect()
+        model = 'USB-2SP2T-DCH'
+        actual = self.connection.Read_ModelName("")[1]
+        if actual != model:
+            raise ValueError(f'Got different model {actual}')
+
+    def disconnect(self):
+        self.connection.Disconnect()
+
+    def ask(self, query):
+        status = self.connection.Send_SCPI(query, "")
+        if status[0] == 1:
+            return str(status[-1])
+        else:
+            raise ValueError('Query was not successful, got 0 status')
+
+    @property
+    def serial_number(self):
+        """ Returns the serial number of the connected switch """
+        return self.connection.Read_SN("")[1]
+
+    @property
+    def CH_A_state(self):
+        """ Returns the serial number of the connected switch """
```

Comment:
In the docstring of the property-getter, also the doc for the property-setter should be included; e.g. 'property that controls....'

Moreover, you might want to check the docstrings, they're all rather similar